File: dotnet/Vaas/src/Vaas/Options/ForUrlOptions.cs
namespace Vaas.Options;

public class ForUrlOptions
{
    public bool UseHashLookup { get; init; } = true;
    public string? VaasRequestId { get; init; }

    public static ForUrlOptions From(VaasOptions options)
    {
        return new ForUrlOptions { UseHashLookup = options.UseHashLookup, VaasRequestId = null };
    }
}

File: typescript/src/options/ForUrlOptions.ts
import { VaasOptions } from "./VaasOptions";

export class ForUrlOptions {
  public useHashLookup: boolean;
  public vaasRequestId?: string;

  constructor(options?: { useHashLookup?: boolean; vaasRequestId?: string }) {
    this.useHashLookup = options?.useHashLookup ?? true;
    this.vaasRequestId = options?.vaasRequestId;
  }

  public static from(options: VaasOptions): ForUrlOptions {
    return new ForUrlOptions({
      useHashLookup: options.useHashLookup,
    });
  }
}
